app/scripts/transaction-manager.js

```diff
@@ -240,17 +256,26 @@ module.exports = class TransactionManager extends EventEmitter {
   //  checks if a signed tx is in a block and
   // if included sets the tx status as 'confirmed'
   checkForTxInBlock () {
-    var signedTxList = this.getFilteredTxList({status: 'signed', err: undefined})
+    var signedTxList = this.getFilteredTxList({status: 'signed'})
     if (!signedTxList.length) return
     signedTxList.forEach((tx) => {
       var txHash = tx.hash
       var txId = tx.id
-      if (!txHash) return
+      if (!txHash) {
+        tx.err = {
+          errCode: 'No hash was provided',
+          message: 'Tx could possibly have not been submitted or an error accrued during signing',
```

Comment:
IN THE MEANWHILE:  We could just make this error something like `We had an error while submitting this transaction, please try again.`.

We shouldn't rely on that for long, but it could let us get on with this PR while fixing that bug separately.